README.md

```diff
@@ -2217,4 +2217,219 @@ A Japanese word meaning "form", relating to karate. Used commonly in programming
+- Medium: What are the benefits of TDD?
+- Hard: What are the downsides of TDD?
+
+### Design
+
+- Easy: What is the difference between design and documentation? How are they similar?
+- Medium: Describe a design process you think will work for a software project.
+- Hard: What concerns do you have to keep in mind when working on a team that is working on several projects at the same time.
+
+## Internet Browsers
+
+There are 3 languages that form what we see when we look at internet browsers.
+
+HTML (Hypertext Markup Language) forms the structure of web pages.
+CSS (Cascading Style Sheets) forms the style.
+JavaScript provides the dynamic and interactive parts.
+
+### JavaScript
+
+JavaScript started as a browser language, but now it is used for a variety of things including: Slack, Facebook Messenger, Web Servers and Applications.
+
+The JavaScript execution engine is Node. No HTML is required, but CSS is.
+
+As JavaScript does not have types, it does not have type safety. Microsoft addressed the issues of this by creating TypeScript, which includes all the capabilities of JavaScript and additionally includes types.
+
+#### A Brief History of JavaScript
+
+Brendan Ike wrote ECMAScript (known widely as JavaScript, although it has no relation to Java) in 14 days. Whilst this is an incredibly impressive achievement, the language was dogged with problems until 2015. Part of the issue was that the problems couldn’t be changed as internet sites depended on it and needed it to have backwards compatibility.
+
+The evolution of evergreen browsers meant that it was possible for a new version, ES6, of JavaScript to be released in 2015. One of the differences is that JavaScript variables used to be symbolised as `var` and now it is `let`, or `const` if the variable shouldn’t change.
+
+### JavaScript Data Types
+
+- String
+- Date
+- Number
+- Boolean
+- Functions
+- null
+- undefined
+- Array
+- Object
+
+### HTML
+
+HTML is not a programming language. It does have classes and IDs for referencing components that programmers wish to reassign. IDs must be assigned only one value where as classes can have multiple values (e.g. if programmers want many aspects to have the same formatting).
+
+HTML and pictures have been combined to create Scalar Vector Graphics (SVG).
+
+The basic format for HTML is:
+
+``` html
+<html>
+<head>
+Put links to dependencies here.
+</head>
+<body>
+Put content you want users to see here.
+</body>
+</html>
+```
+
+#### HTML Symbols
+
+| Symbol | Definition |
+| ------ | ---------- |
+| `<p>`  | paragraph  |
+| `<b>`  |   body     |
+| `<div>` | defines a division or section |
+| `<!-- -->` | comments in the middle of symbol |
+
+#### HTML vs Markdown 
+
+| Markdown | HTML |
+| -------- | ---- |
+|  `#`  | `<h1>` |
+|  `-`  | `<ul> <li> text goes here </li> </ul>` |
+|  `1.` | `<ol> <li> text goes here </li> </ol>` |
+| backtick | `<pre>` |
+| ` ```java | `<pre><code></code></pre>` |
```

Comment:
I don't think this renders quite right. Maybe

```suggestion
| ` ```java ` | `<pre><code></code></pre>` |
```